test/e2e/pythonruntime_test.go

```diff
@@ -0,0 +1,435 @@
+
+	duration := time.Since(startTime)
+	testingT.Logf("Test completed successfully: duration %s", duration)
+}
+
+// TestRunPythonRuntimeSandboxWarmpool tests that we can run the Python runtime inside a Sandbox.
+func TestRunPythonRuntimeSandboxWarmpool(testingT *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	testContext := framework.NewTestContext(testingT)
+
+	ns := &corev1.Namespace{}
+	ns.Name = fmt.Sprintf("python-sandbox-warmpool-test-%d", time.Now().UnixNano())
+	require.NoError(testingT, testContext.CreateWithCleanup(testingT.Context(), ns))
+
+	startTime := time.Now()
+
+	// Apply python runtime sandbox template, warmpool manifests
+	manifest := fmt.Sprintf(templateManifest, getImagePrefix(), getImageTag())
+	sandboxTemplate, err := sandboxTemplateFromManifest(manifest)
+	require.NoError(testingT, err)
+	sandboxTemplate.Namespace = ns.Name
+	require.NoError(testingT, testContext.CreateWithCleanup(testingT.Context(), sandboxTemplate))
+
+	sandboxWarmpool, err := sandboxWarmpoolFromManifest(warmPoolManifest)
+	require.NoError(testingT, err)
+	sandboxWarmpool.Namespace = ns.Name
+	require.NoError(testingT, testContext.CreateWithCleanup(testingT.Context(), sandboxWarmpool))
+
+	sandboxWarmpoolID := types.NamespacedName{
+		Namespace: ns.Name,
+		Name:      "python-warmpool",
+	}
+
+	// Wait for the warmpool to be ready
+	require.NoError(testingT, testContext.WaitForWarmPoolReady(testingT.Context(), sandboxWarmpoolID))
+
+	// Apply python runtime sandbox claim manifest
+	sandboxClaim, err := sandboxClaimFromManifest(claimManifest)
+	require.NoError(testingT, err)
+	sandboxClaim.Namespace = ns.Name
+	require.NoError(testingT, testContext.CreateWithCleanup(testingT.Context(), sandboxClaim))
+
+	sandboxID := types.NamespacedName{
+		Namespace: ns.Name,
+		Name:      "python-sandbox-claim",
+	}
+
+	require.NoError(testingT, testContext.WaitForSandboxReady(testingT.Context(), sandboxID))
+
+	// Get the SandboxClaim to extract the sandbox name
+	sandbox, err := testContext.GetSandbox(ctx, sandboxID)
+	require.NoError(testingT, err)
+
+	sandboxName, _, err := unstructured.NestedString(sandbox.Object, "metadata", "annotations", "agents.x-k8s.io/pod-name")
+	require.NoError(testingT, err)
+	testingT.Logf("DEBUG: Extracted SandboxName from Sandbox: sandboxName - %s", sandboxName)
+
+	podID := types.NamespacedName{
+		Namespace: ns.Name,
+		Name:      sandboxName,
+	}
+
+	// Run the tests on the pod
+	require.NoError(testingT, runPodTests(ctx, testingT, testContext, podID))
+
+	duration := time.Since(startTime)
+	testingT.Logf("Test completed successfully: duration-%s", duration)
+}
+
+// runPodTests runs the health check, root endpoint, and execute endpoint tests on the given pod.
+func runPodTests(ctx context.Context, testingT *testing.T, testContext *framework.TestContext, podID types.NamespacedName) error {
+	testContext.Helper()
+	pollDuration := 200 * time.Millisecond
+	for {
```

Comment:
Would it work if we move the port-foward outside of the for loop ? wdyt ? I fear the test could get flaky if we get into an issue of `Address already in use` because we do port-forward inside the for loop